common/src/main/java/com/tencent/rss/common/ShuffleIndexResult.java

```diff
@@ -18,22 +18,47 @@
 
 package com.tencent.rss.common;
 
+import com.google.common.collect.Lists;
+import com.google.common.primitives.Bytes;
+import java.util.List;
+
 public class ShuffleIndexResult {
-  private final byte[] indexData;
+  private final List<ShuffleIndexDataItem> shuffleIndexDataItems;
 
   public ShuffleIndexResult() {
-    this(new byte[0]);
+    this.shuffleIndexDataItems = Lists.newArrayList();
+  }
+
+  public ShuffleIndexResult(ShuffleIndexDataItem shuffleIndexDataItem) {
+    this.shuffleIndexDataItems = Lists.newArrayList();
+    if (!shuffleIndexDataItem.isEmpty()) {
+      shuffleIndexDataItems.add(shuffleIndexDataItem);
+    }
   }
 
-  public ShuffleIndexResult(byte[] bytes) {
-    this.indexData = bytes;
+  public ShuffleIndexResult(List<ShuffleIndexDataItem> indexDataItems) {
+    this.shuffleIndexDataItems = Lists.newArrayList();
+    for (ShuffleIndexDataItem item : indexDataItems) {
+      if (!item.isEmpty()) {
+        shuffleIndexDataItems.add(item);
+      }
```

Comment:
need we log warn some message when item is empty.

Reply:
OK